### DISCUSSION TITLE
Custom Theme my Cursor Position is Wrong

### DISCUSSION THREAD

**Discussion by hackdefendr** [NuSkooler/enigma-bbs#469] [Q&A]
First, hello everyone! I really hope someone can help me figure out the basics as this is my second attempt at setting up an Enigma BBS and I'm right back where I ended up last time so I thought I should reach out for help on some basic stuff.

My setup:
- VM with 2 vCPU, 8G Ram, 120G storage
- Ubuntu Server 22.0.4
- Latest Enigma via installer
- Custom theme from Luciano
- Running both Telnet and SSH (plan on added Websocket)

Client side (Macbook Pro M1x):
- PabloDraw
- SyncTerm

I purchased a custom theme from Luciano and chopped up the files so that they used the same file names and structure as the default theme that comes with the code. I made sure to crop everything to 80x25 and then placed any menu items and codes where they could fit.  I hoped that things like cursor position would just work but they do not for me.

I'm using SyncTerm for testing so things display properly and right at the start the Login, Apply, Logout buttons are not where I think I placed the codes in the art files (see the image below). I need help understanding why this is and why cursor placement never works in my custom theme the same as it does in the default theme.

<img width="1250" alt="Screenshot 2023-03-26 at 5 22 56 PM" src="https://user-images.githubusercontent.com/2224217/227808573-973dba5f-2ec2-4e6f-9e47-79c0078f6590.png">

The empty red box at the top is where I think I placed the code for the login menu to go.

I definitely have read the documentation site, but something just isn't clicking in several areas. I understand this may be a huge thing to take on, however I am very technical so I am hoping that someone will describe things differently and I will start understanding it all. It is like I am staring at the words but they aren't translating in so my brain gets it.

Regards,
Jeff

**Comment by NuSkooler** [comment]
Some questions:
1. Did you update your `theme.hjson` to accommodate your new layout/art?
2. 80x25 is too much for a number of terminals, so 79x24 is a common work around. SyncTERM in particular reports it's terminal size wrong if the status bar is displayed, so that may be a issue as well.

Make sure the artwork is really the size you think as well. Sometimes PabloDraw can save some extra "blank" text under what you think is the correct size. Moebius ANSI editor allows you to explicitly set the size, which can be helpful as well.

Let us know if that helps!

**Reply by cognitivegears** [reply]
@hackdefendr also, we just checked in a change that might affect this - can you update to the latest and re-test? Otherwise if you don't mind please create an issue and include the ANSI file (easy way to do that is to base64 encode it so you can paste it into the issue.) Thanks!

**Comment by john-clark** [comment]
I know this is old but the reason I am commenting on this is because I was just looking at this. I was trying to figure out how to change the font capitalization when you arrow around on the menu. I noticed that SSH skips this screen, so this screen only working in telnet. However I have moved around the %VM1%VM1 variable around the screen with PabloDraw and did not have this issue.

@hackdefendr It would be good for you to take a screenshot of your pablodraw screen as well as the syncterm. It looks like the %VM1%VM1 is in the wrong place in your MATRIX.ANS file.
